.github/workflows/main-build.yml

```diff
@@ -102,12 +110,204 @@ jobs:
+
+  # job to run the e2e integration tests
+  run-batch-job:
+    runs-on: ubuntu-latest
+    needs: [build, create-test-ref, get-testing-suites]
+    strategy:
+      fail-fast: false
+      matrix: ${{ fromJson(needs.get-testing-suites.outputs.test-case-batch-key) }}
+
+    steps:
+      # required for versioning
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          path: aws-otel-java-instrumentation
+
+      - name: Set up JDK 11
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'zulu'
+          java-version: '11'
+
+      - name: Set up terraform
+        uses: hashicorp/setup-terraform@v2
+        with:
+          terraform_version: "~1.4"
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+          role-to-assume: ${{ secrets.JAVA_INSTRUMENTATION_INTEG_TEST_ARN}}
+          aws-region: us-west-2
+          # 4 hours
+          role-duration-seconds: 14400
+
+      - name: Checkout testing framework
+        uses: actions/checkout@v3
+        with:
+          repository: ${{ env.TESTING_FRAMEWORK_REPO }}
+          path: testing-framework
+          ref: ${{ needs.create-test-ref.outputs.testRef }}
+
+      - name: create test-case-batch file
+        run: |
+          jsonStr='${{ needs.get-testing-suites.outputs.test-case-batch-value }}'
+          jsonStr="$(jq -r '.${{ matrix.BatchKey }} | join("\n")' <<< "${jsonStr}")"
+          echo "$jsonStr" >> testing-framework/terraform/test-case-batch
+          cat testing-framework/terraform/test-case-batch
+
+      - name: Get TTL_DATE for cache
+        id: date
+        run: echo "ttldate=$(date -u -d "+7 days" +%s)" >> $GITHUB_OUTPUT
+
+      - name: run tests
+        run: |
+          export TTL_DATE=${{ steps.date.outputs.ttldate }} 
+          export TF_VAR_JAVA_AUTO_INSTRUMENTATION_TAG=${{ needs.build.outputs.java_agent_tag }}
+          export TF_VAR_JAVA_AUTO_INSTRUMENTATION_REPOSITORY=${{ env.STAGING_ECR }}
+          export DDB_BATCH_CACHE_SK=${{ needs.build.outputs.java_agent_tag }}
+          cd testing-framework/terraform
+          make execute-batch-test
+
+      - name: output cache misses
+        if: ${{ failure() }}
+        run: |
+          export DDB_BATCH_CACHE_SK=${{ needs.build.outputs.java_agent_tag }}
+          cd testing-framework/terraform
+          make checkCacheHits
+
+      # This is here just in case workflow cancel
+      # We first kill terraform processes to ensure that no state
+      # file locks are being held from SIGTERMS dispatched in previous
+      # steps.
+      - name: Destroy resources
+        if: ${{ cancelled() }}
+        shell: bash {0}
+        run: |
+          ps -ef | grep terraform | grep -v grep | awk '{print $2}' | xargs -n 1 kill
+          cd testing-framework/terraform
+          make terraformCleanup
+
+
+  build_Images_For_Testing_Sample_App_With_Java_Agent:
```

Comment:
don't you need to update the steps that rely on this one? i.e.: don't you need to update the `needs` property of `test_Spring_App_With_Java_Agent` for instance?

Reply:
thanks @rapphil. Updated.

Reply:
What exactly is this step building and pushing?

Reply:
this step is building and publishing images (sample apps and java agent) required for testing performed in these subsequent steps:
```
test_Spark_AWS_SDK_V1_App_With_Java_Agent
test_Spark_App_With_Java_Agent
test_Spring_App_With_Java_Agent
```

Reply:
Why does it have a need on `run-batch-job`? Isn't that the integration test for the java agent image?

Reply:
@bryan-aguilar I added `run-batch-job` as a need to ensure that e2e integ-testing is passing before reaching this step and start publishing new java agent and sample-app images. 
There is no hard dependency. we can run these jobs in parallel but I provided this to run e2e integ-tests and sample-app testing in series, so that if there is a failure in e2e tests then we can completely avoid running these sample-app testing.

Reply:
I think there is little benefit in doing in serial. lets do in parallel.

Reply:
@rapphil . I have updated accordingly. Thanks!